server/server.go

```diff
@@ -137,12 +139,40 @@ func (s *Server) SyncStatus(instID flux.InstanceID, ref string) (res []string, e
 }
 
 // LogEvent stores an event in the instance's history.
-func (s *Server) LogEvent(instID flux.InstanceID, event history.Event) error {
+func (s *Server) LogEvent(instID flux.InstanceID, e history.Event) error {
 	helper, err := s.instancer.Get(instID)
 	if err != nil {
 		return errors.Wrapf(err, "getting instance")
 	}
-	return helper.LogEvent(event)
+
+	// Log event in history first. This is less likely to fail
+	err = helper.LogEvent(e)
+	if err != nil {
+		return errors.Wrapf(err, "logging event")
+	}
+
+	// If this is a release
+	if e.Type == history.EventRelease {
+		// Casting to ReleaseEventMetadata
+		bytes, err := json.Marshal(e.Metadata)
```

Comment:
OK I think I see. This is the only place having something other than `map[string]interface{}` is necessary.

But neither would it hurt to implement `func (e *Event) UnmarshalJSON` so that it constructs the real types, since they would marshal back to JSON for the database. And that would avoid this extra trip through encoding/json.